aiida/cmdline/commands/cmd_daemon.py

```diff
@@ -158,7 +158,7 @@ def logshow():
 
     try:
         currenv = get_env_with_venv_bin()
-        process = subprocess.Popen(['tail', '-f', client.daemon_log_file], env=currenv)
+        process = subprocess.Popen(['tail', '-f', client.daemon_log_file], env=currenv)  # pylint: disable=consider-using-with
```

Comment:
Instead of disabling, couldn't we actually not better use a with statement here?

Reply:
It wasn't quickly apparent to me how to do this and keep the same behaviour. i.e. is the same?

```
currenv = get_env_with_venv_bin()
with subprocess.Popen(['tail', '-f', client.daemon_log_file], env=currenv) as process:
   try:
     process.wait()
   except KeyboardInterrupt:
     process.kill()
```

probably not 🤷 


same with the other `disable=consider-using-with`

Reply:
Fair enough. I would be tempted to do:
```
currenv = get_env_with_venv_bin()
try:
    with subprocess.Popen(['tail', '-f', client.daemon_log_file], env=currenv) as process:
        process.wait()
except KeyboardInterrupt:
    process.kill()
```
But I am not sure the try-except is even necessary when using the context manager. Keyboard interrupt should just kill it and the context manager should clean it up properly. Anyway, indeed fine to keep this as is as it is not a trivial change